presto-native-execution/presto_cpp/main/common/Configs.cpp

```diff
@@ -142,21 +142,36 @@ ConfigBase::ConfigBase()
     : config_(std::make_unique<velox::core::MemConfig>()) {}
 
 void ConfigBase::initialize(const std::string& filePath) {
-  // See if we want to create a mutable config.
-  auto values = util::readConfig(fs::path(filePath));
+  std::unordered_map<std::string, std::string> configFromFile =
+      util::readConfig(fs::path(filePath));
   filePath_ = filePath;
-  checkRegisteredProperties(values);
+  checkRegisteredProperties(configFromFile);
 
+  // See if we want to create a mutable config.
   bool mutableConfig{false};
-  auto it = values.find(std::string(kMutableConfig));
-  if (it != values.end()) {
+  auto it = configFromFile.find(std::string(kMutableConfig));
+  if (it != configFromFile.end()) {
     mutableConfig = folly::to<bool>(it->second);
   }
 
+  std::unordered_map<std::string, std::string> config;
+  for (auto property = registeredProps_.begin();
+       property != registeredProps_.end();
+       property++) {
+    if (property->second.hasValue()) {
+      config.insert({property->first, property->second.value()});
+    }
+  }
+  // Config from file would overwrite the value from registeredProps_.
+  for (auto property = configFromFile.begin(); property != configFromFile.end();
+       property++) {
+    config.insert({property->first, property->second});
```

Comment:
"insert" inserts element(s) into the container, if the container doesn't already contain an element with an equivalent key.

Which means it won't overwrite the existing ones. Have you tested it?

Reply:
I just noticed that ConfigTest would fail with current version, after changing, it would pass.
Thanks for catching the issue!